src/game.cpp

```diff
@@ -2464,6 +2469,36 @@ void Game::processKeyInput()
 		toggleFast();
 	} else if (wasKeyDown(KeyType::NOCLIP)) {
 		toggleNoClip();
+	} else if (wasKeyDown(KeyType::MUTE)) {
+		float volume = g_settings->getFloat("sound_volume");
+		if(volume < 0.001) {
+			g_settings->setFloat("sound_volume", 1.0);
+			m_statustext = gettext(L"Volume changed to 100%");
+		} else {
+			g_settings->setFloat("sound_volume", 0.0);
+			m_statustext = gettext(L"Volume changed to 0%");
+		}
+		runData.statustext_time = 0;
+	} else if (wasKeyDown(KeyType::INC_VOLUME)) {
+		float volume = g_settings->getFloat("sound_volume");
+		float new_volume = volume + 0.1;
+		if(new_volume > 1)
```

Comment:
How about:
`float volume = rangelim(g_settings->getFloat("sound_volume") + 0.1, 0, 1);`
to eliminate any bad input from the settings?